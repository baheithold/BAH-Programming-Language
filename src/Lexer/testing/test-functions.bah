# This file tests valid functions

function isOdd(var num) {
    return num % 2 == 0;
}

var odd? = isOdd(5);
