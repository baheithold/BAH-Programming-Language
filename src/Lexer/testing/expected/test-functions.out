FUNCTION_KEYWORD
ID_TYPE isOdd
OPAREN
VAR_KEYWORD
ID_TYPE num
CPAREN
OBRACE
RETURN_KEYWORD
ID_TYPE num
MODULO_BINARY
INTEGER_TYPE 2
EQUALS_COMPARATOR
INTEGER_TYPE 0
SEMICOLON
CBRACE
VAR_KEYWORD
ID_TYPE odd?
ASSIGN_BINARY
ID_TYPE isOdd
OPAREN
INTEGER_TYPE 5
CPAREN
SEMICOLON
END_OF_INPUT
